sway/commands/move.c

```diff
@@ -37,19 +50,33 @@ static struct sway_container *output_in_direction(const char *direction,
 		{ "left", WLR_DIRECTION_LEFT },
 		{ "right", WLR_DIRECTION_RIGHT },
 	};
+
+	enum wlr_direction *direction = NULL;
+
 	for (size_t i = 0; i < sizeof(names) / sizeof(names[0]); ++i) {
-		if (strcasecmp(names[i].name, direction) == 0) {
-			struct wlr_output *adjacent = wlr_output_layout_adjacent_output(
+		if (strcasecmp(names[i].name, direction_string) == 0) {
+			direction = &names[i].direction;
```

Comment:
Another nitpick: You could add a `break` here to stop iterating the loop once you've found the direction.